lib/classes/ConfigSchemaHandler.js

```diff
@@ -0,0 +1,76 @@
+'use strict';
+
+const Ajv = require('ajv');
+const schema = require('../configSchema');
+const ajvErrorMessageBuilder = require('../utils/ajvErrorMessageBuilder');
+
+const functionNamePattern = '^[a-zA-Z0-9-_]+$';
+
+class ConfigSchemaHandler {
+  constructor(serverless) {
+    this.serverless = serverless;
+    this.schema = schema;
+
+    deepFreeze(this.schema.properties.service);
+    deepFreeze(this.schema.properties.plugins);
+    deepFreeze(this.schema.properties.resources);
```

Comment:
I think `resources` are allowed to have different forms in different providers.. we probably should just freeze that it's an object, but do not freeze internal structure of it.

Reply:
So what you mean is change to this:

```suggestion
    Object.freeze(this.schema.properties.resources);
```

correct?

Reply:
Yes, thank you!